### PR TITLE
Update UI/UX practice

### PR DESCRIPTION
**Pull request by daoanhhuy26012001**
- [x] Remove `useClickOutside`
- [x] Change data connection to remove `useEffect()`
- [x] Use a modal for a popup
- [x] Add back button for wishlist

### CODE REVIEW

File: ReactJS/practice-one/src/App.jsx [hunk 1]
```diff
@@ -158,12 +161,21 @@ function App() {
...
-          <ProductDetails
-            product={selectedProduct}
-            setSelectedProduct={setSelectedProduct}
-          />
+        <div
+          className="modal"
+          id="exampleModal"
+          tabIndex="-1"
+          aria-labelledby="exampleModalLabel"
+          aria-hidden="true">
```
Comment (phattruong-agilityio) [inline ReactJS/practice-one/src/App.jsx:169]: always "true"??

File: ReactJS/practice-one/src/App.jsx [hunk 2]
```diff
@@ -158,12 +161,21 @@ function App() {
...
       {isOpenProductDetailPopup && (
-        <div className="popup">
-          {/* Product details popup */}
-          <ProductDetails
-            product={selectedProduct}
-            setSelectedProduct={setSelectedProduct}
-          />
+        <div
+          className="modal"
+          id="exampleModal"
```
Comment (phattruong-agilityio) [inline ReactJS/practice-one/src/App.jsx:166]: example Modal?

File: ReactJS/practice-one/src/App.jsx [hunk 3]
```diff
@@ -158,12 +161,21 @@ function App() {
...
+          tabIndex="-1"
+          aria-labelledby="exampleModalLabel"
```
Comment (phattruong-agilityio) [inline ReactJS/practice-one/src/App.jsx:168]: the same here

File: ReactJS/practice-one/src/App.jsx [hunk 4]
```diff
@@ -124,16 +116,27 @@ function App() {
...
-          />
-        )}
       </div>
+      {/* Wish list component */}
+      {wishListVisible && (
+        <div
+          className="modal"
+          id="exampleModal"
+          tabIndex="-1"
+          aria-labelledby="exampleModalLabel"
```
Comment (phattruong-agilityio) [inline ReactJS/practice-one/src/App.jsx:126]: example??

Comment (daoanhhuy26012001) [reply ReactJS/practice-one/src/App.jsx]: Done

File: ReactJS/practice-one/src/App.jsx (hunk 3 above)
Comment (daoanhhuy26012001) [reply ReactJS/practice-one/src/App.jsx]: Done

File: ReactJS/practice-one/src/App.jsx (hunk 2 above)
Comment (daoanhhuy26012001) [reply ReactJS/practice-one/src/App.jsx]: Done

File: ReactJS/practice-one/src/App.jsx (hunk 1 above)
Comment (daoanhhuy26012001) [reply ReactJS/practice-one/src/App.jsx]: Done

**Pull request merged**